pkg/cassandra/config.go

```diff
@@ -50,54 +63,75 @@ func isCassandra4(version string) bool {
 }
 
 func (c config) MarshalJSON() ([]byte, error) {
-	config := make(map[string]interface{})
+	cassConfig := make(map[string]interface{})
 
 	// Even though we default to Cassandra's stock defaults for num_tokens, we need to
 	// explicitly set it because the config builder defaults to num_tokens: 1
-	if c.NumTokens == nil {
+	if c.CassandraYaml.NumTokens == nil {
 		if isCassandra4(c.cassandraVersion) {
-			c.NumTokens = pointer.Int(16)
+			c.CassandraYaml.NumTokens = pointer.Int(16)
 		} else {
-			c.NumTokens = pointer.Int(256)
+			c.CassandraYaml.NumTokens = pointer.Int(256)
 		}
 	}
 
-	config["cassandra-yaml"] = c.CassYamlIR
+	cassConfig["cassandra-yaml"] = c.CassYamlIR
 
 	if c.jvmOptions.InitialHeapSize != nil || c.jvmOptions.MaxHeapSize != nil || c.jvmOptions.HeapNewGenSize != nil {
 		if isCassandra4(c.cassandraVersion) {
-			config["jvm-server-options"] = c.jvmOptions
+			cassConfig["jvm-server-options"] = c.jvmOptions
 		} else {
-			config["jvm-options"] = c.jvmOptions
+			cassConfig["jvm-options"] = c.jvmOptions
 		}
 	}
 
 	// Many config-builder templates accept a parameter called "additional-jvm-opts", e.g the jvm-options or the
 	// cassandra-env-sh templates; it is safer to include this parameter in cassandra-env-sh, so that we can guarantee
 	// that our options will be applied last and will override whatever options were specified by default.
 	if c.additionalJvmOptions != nil {
-		config["cassandra-env-sh"] = map[string]interface{}{
+		cassConfig["cassandra-env-sh"] = map[string]interface{}{
 			"additional-jvm-opts": c.additionalJvmOptions,
 		}
 	}
 
-	return json.Marshal(&config)
+	return json.Marshal(&cassConfig)
 }
 
 // NewIRYaml returns a new Cassandra Yaml with all internal fields set to their mandatory values. For example, StartRpc must always be set false,
 // NewIRYaml ensures this happens.
 func NewIRYaml(cassAPIYaml api.CassandraYaml) CassYamlIR {
```

Comment:
Just FYI. This function should not be exported either.

Reply:
👍